### ISSUE TITLE
MechJeb debug arrows spamming error log

### ISSUE THREAD

**Issue by MOARdV** [MuMech/MechJeb2#537]
I found this on dev build 399.  The debug arrows appear to be spamming the error log as thus:

> [ERR 08:30:42.307] MechJeb module MechJebModuleDebugArrows threw an exception in OnUpdate: System.NullReferenceException: Object reference not set to an instance of an object
>   at MuMech.MechJebModuleDebugArrows.OnUpdate () [0x00000] in <filename unknown>:0 
>   at MuMech.MechJebCore.Update () [0x00000] in <filename unknown>:0 

The NREs start after launch (not present while the craft is still on the launch pad, prior to staging).

I've also noticed that the arrows appear when warping (specifically, one yellow/gold arrow), and that when leaving warp I can see it floating somewhere within physics range of the ship.  During one launch, during circularization, the arrow (I think) flew past the ship, making me think for a moment that I was going to have a collision with debris.

I do not have any of the arrows checked in the UI.

There's also a single NRE coming from FlameoutSafetyThrottle right after launch.

**Comment by sarbian** [comment]
I made some change than went in the last build. It should fix the various problems with the arrows

**Comment by MOARdV** [comment]
That fixed the spamming, but I now see the MET timer flashing yellow a lot, and the arrows are still visible:

http://imageshack.com/a/img911/6324/jsngqI.jpg
http://imageshack.com/a/img631/1821/Ho3hZo.jpg

Tested with MJ dev build 400.

**Comment by sarbian** [comment]
I don't understand how they can be visible if you disabled them ....

**Comment by MOARdV** [comment]
I don't understand either.  Here's a screenshot showing the arrow (pointing retrograde) with the Attitude Adjustment window open - no arrows are checked.  Unless there's another place in the UI where I have to disable it?  What may not be visible in this screen shot is that there is also a red "ball" at the base of the yellow arrow.

http://imageshack.com/a/img911/5168/DkYRtV.jpg

**Comment by MOARdV** [comment]
One more data point from some additional playing / testing.  It appears to be limited to that particular craft.  The satellites / probes / manned vessels I've tried don't seem to be generating spurious arrows.

**Comment by MOARdV** [comment]
More details, including logs and the persistent.sfs (which may not help, since it'll require multiple mods to load the affected parts - HGR_Redux, Station Parts Expansion, KOSMOS, possibly RLA Stockalike and others):

https://www.dropbox.com/s/w6y9oafazf6pi9h/KSP-bug-MJ.zip?dl=0

I launched a craft "Highline Comms Array", which consisted of a stock 1.25m inline probe, RCS fuel tanks, and various and sundry other parts.  I flew it to rendezvous with "Highline Station", which consisted of a 2,5m inline probe and various other components.  I am using an MM patch to add MechJeb to all command pods / probe cores.

After docking the comms array to the station, I separated the Highline Comms Array probe / fuel unit to fly it to the other end of the comms array, where it would dock with a spare fuel tank, separate it from the station, and deorbit.

I separate the orbital tug (probe core) in the log file at "Switching to Vessel Highline Comms Array Probe".  It is here that I start seeing the null reference exceptions.  Once the craft docks with the extra fuel tank (and connects to the station again), the exceptions go away.  When it separates again (becoming the Highline Comms Array Probe once more), the exceptions return.  The null deference exceptions are only present when the Probe is under control.  I don't know if this helps.
